core/src/stored/backends/unix_tape_device.cc

```diff
@@ -49,6 +52,19 @@
 #include "generic_tape_device.h"
 #include "unix_tape_device.h"
 
+namespace {
+static constexpr size_t size_increment = 1024 * 1024;
```

Comment:
Does an anonymous namespace not already serve like the static specifier?

Reply:
My thought was that putting stuff in the global namespace isn't great (even though it is static), but putting it in the storagedaemon namespace wouldn't be great either.
And that's how it ended up in an anonymous namespace.

Reply:
might be a bit too cute but what about
```
class bufsizes
{
public:
bufsizes() = default;
bufsizes(std::size_t initial_size) : current_size{round_up(initial_size)} {}
bufsizes end() { return bufsizes{}; }
bufsizes begin() { return *this; }
bufsizes& operator++() {
  current_size = next(current_size);
  return *this;
}

std::size_t operator*() {
    return current_size;
}

friend bool operator!=(const bufsizes& lhs, const bufsizes& rhs) {
    return lhs.current_size != rhs.current_size;
}

private:
static constexpr std::size_t increment = 1024 * 1024;
static constexpr std::size_t max_size = 16 * increment;
std::size_t current_size{next(max_size)};
static constexpr std::size_t round_up(std::size_t val) {
    return ((val + increment - 1) / increment) * increment;
}

static constexpr std::size_t next(std::size_t val) {
    return 2 * val;
}
};
```

Reply:
We can even simplify more: it is sufficient to have a `std::vector<size_t> bufsizes(size_t start)` that returns the sizes to try, no need for a new type...

Reply:
What about the following implementation?
```
std::vector<std::size_t> bufsizes(std::size_t count) {
  constexpr std::size_t mb = 1024 * 1024;
  std::vector<std::size_t> sizes{1 * mb, 2 * mb, 4 * mb, 8 * mb, 16 * mb};
  sizes.erase(std::remove_if(sizes.begin(), sizes.end(),
                             [count](size_t s) { return s <= count; }),
              sizes.end());
  return sizes;
}
```

Reply:
That also works.  Its a shame that it allocated memory, but we can just replace it by span once C++20 lands and we touch the code again.

Reply:
This is splitting hairs but i think we can just use
```
std::erase(std::begin(sizes), std::upper_bound(sizes.begin(), sizes.end(), count));
```
since we know the vector is sorted.

Reply:
Yep, that's definitely better.

Concerning the allocation I don't think it is a problem.
1. we're on the error-path
2. we're rewinding a tape once, which is probably one of the slowest operations possible

If we really want to get rid of the allocation we can use a `static constexpr std:array<size_t>` to iterate over and then just `if (x < y) continue;` within the loop.